compiler/model/metamodel.ts

```diff
@@ -376,6 +376,7 @@ export class UrlTemplate {
 export class Model {
   _info?: {
     version: string
+    hash: string
```

Comment:
We're doing something similar in the [`download-schema` task](https://github.com/elastic/elastic-client-generator-java/blob/38da213659c72b1af58cd308aa0609c837070590/generator/build.gradle.kts#L65) in the Java client generator, but also add the branch name, which could be useful in some contexts.

The resulting metadata would be:

```typescript
export class Model {
  _info?: {
    version: string
    git: {
        revision: string
        branch: string
    }
    title: string
    ...
```

Reply:
In this case, version and branch are the same thing.

Reply:
> In this case, version and branch are the same thing.

Hmm... It should be different for `main` and maybe also for `7.x`. Here's what I have as a result of the Java generator's `download-schema`:

```json
  "_info" : {
    "license" : {
      "name" : "Apache 2.0",
      "url" : "https://github.com/elastic/elasticsearch-specification/blob/master/LICENSE"
    },
    "title" : "Elasticsearch Request & Response Specification",
    "version" : "7.13.0",
    "git" : {
      "branch" : "main",
      "revision" : "d59679ae28f894729f11a206042ba24533a5624c"
    }
  },
```

Reply:
In terms of use cases, having the branch name can be useful for example for tools that would compare an existing schema with the current revision from the same branch (API change log, backwards compatibility check, etc)

Reply:
Yes, but in our case, we are not tracking patch numbers, and the branch name is the "version" that we are actually tracking.

Eg, in the `main` branch we are developing the `main` version of the specification, the same goes for `7.x` and `7.14`.